ipywidgets/widgets.py

```diff
@@ -26,59 +27,64 @@ def renamed(self, name):
         else:
             obj = self
         obj.name = name
+        if obj.description is None:
+            obj.description = name
         return obj
 
 
 class RangeWidget(StaticWidget):
     """Range (slider) widget"""
-    slider_html = ('<b>{name}:</b> <input type="range" name="{name}" '
+    slider_html = ('<b>{description}</b> = '
+                   '<label id="{name}' + '_label" for="{name}">{default}</label>'
+                   '<input type="range" id="{name}" name="{name}" '
                    'min="{range[0]}" max="{range[1]}" step="{range[2]}" '
                    'value="{default}" style="{style}" '
-                   'oninput="interactUpdate(this.parentNode);" '
-                   'onchange="interactUpdate(this.parentNode);">')
+                   'oninput="interactUpdate(this);" '
```

Comment:
A bit more general, I get the parent node inside interactUpdate() instead. I need the element instead of the parent node here to retrieve the element id and value.